File: src/main/java/com/example/AI_Study_Planer/dto/response/Quiz/QuizDetailResponse.java
package com.example.AI_Study_Planer.dto.response.Quiz;

import com.example.AI_Study_Planer.enums.Quiz.Difficulty;
import lombok.Data;

import java.util.List;

@Data
public class QuizDetailResponse {

    private String id;

    private String title;

    private String description;

    private Difficulty difficulty;

    private Integer durationMinutes;

    private Integer totalQuestions;

    private String category;

    private Boolean aiRecommended;

    private List<QuizQuestionResponse> questions;
}

File: src/types/Response/Quiz/QuizDetailResponse.ts
import type { Difficulty } from "@/types/Enums";
import type { QuizQuestionResponse } from "./QuizQuestionResponse";

export interface QuizDetailResponse {
  id: string;

  title: string;

  description: string | null;

  difficulty: Difficulty;

  durationMinutes: number | null;

  totalQuestions: number;

  aiRecommended: boolean | null;

  category: string | null;

  questions: QuizQuestionResponse[];
}
